src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -33,162 +43,465 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+            s_localizableMessageMethodTypeParameterMoreMeaningfulName,
+            DiagnosticCategory.Naming,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false,
+            description: s_localizableDescription,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1704",
+            customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            FileParseRule,
+            AssemblyRule,
+            NamespaceRule,
+            TypeRule,
+            VariableRule,
+            MemberRule,
+            MemberParameterRule,
+            DelegateParameterRule,
+            TypeTypeParameterRule,
+            MethodTypeParameterRule,
+            AssemblyMoreMeaningfulNameRule,
+            NamespaceMoreMeaningfulNameRule,
+            TypeMoreMeaningfulNameRule,
+            MemberMoreMeaningfulNameRule,
+            MemberParameterMoreMeaningfulNameRule,
+            DelegateParameterMoreMeaningfulNameRule,
+            TypeTypeParameterMoreMeaningfulNameRule,
+            MethodTypeParameterMoreMeaningfulNameRule);
+
         public override void Initialize(AnalysisContext analysisContext)
-#pragma warning restore RS1025 // Configure generated code analysis
         {
             analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext compilationStartContext)
+        {
+            var dictionaries = ReadDictionaries();
+            var projectDictionary = CodeAnalysisDictionary.CreateFromDictionaries(dictionaries.Concat(_mainDictionary));
+
+            compilationStartContext.RegisterOperationAction(AnalyzeVariable, OperationKind.VariableDeclarator);
+            compilationStartContext.RegisterCompilationEndAction(AnalyzeAssembly);
+            compilationStartContext.RegisterSymbolAction(
+                AnalyzeSymbol,
+                SymbolKind.Namespace,
+                SymbolKind.NamedType,
+                SymbolKind.Method,
+                SymbolKind.Property,
+                SymbolKind.Event,
+                SymbolKind.Field,
+                SymbolKind.Parameter);
+
+            IEnumerable<CodeAnalysisDictionary> ReadDictionaries()
+            {
+                var fileProvider = AdditionalFileProvider.FromOptions(compilationStartContext.Options);
+                return fileProvider.GetMatchingFiles(@"(?:dictionary|custom).*?\.(?:xml|dic)$")
+                    .Select(CreateDictionaryFromAdditionalText)
+                    .Where(x => x != null)
+                    .ToList();
+
+                CodeAnalysisDictionary CreateDictionaryFromAdditionalText(AdditionalText additionalFile)
```

Comment:
Could be marked as `static`.

Reply:
Depends if we pass in parameters or capture variables